courses.py

```diff
@@ -1,68 +1,89 @@
+
+"""
+Assumptions:
+    All subject names are 4-letter acronyms
+    All course numbers are 4-digit numbers
+    If a new subject is listed, the format will be ABCD1234 (i.e. no spaces between name and number)
+    If a subject number is listed without a course subject acronym, it's subject is the most recently seen subject:
+        (ex. 'ABCD1000 2000' is equivalent to 'ABCD1000 ABCD2000')
+"""
+
+
+def parse_courses():
+    """
+    Parses the COURSES string into a dictionary.
+    :return: A dictionary whose keys are course name acronyms (ex: 'COMP')
+             and the values are arrays of course numbers (as a numeric string, ex: ['2500', '2501', '2510'])
+    """
+
+    subjects = {}
```

Comment:
See https://docs.python.org/2/library/collections.html#collections.OrderedDict for keeping the classes in the original ordering.